Peaky/IParametrizedTestCases.cs

```diff
@@ -0,0 +1,7 @@
+namespace Peaky
+{
+    public interface IParameterizedTestCases : IPeakyTest
```

Comment:
I don't think this needs to be a public interface.

Reply:
this is a feature, how will the test developer expose the fact that the test exposes parameterized test cases?

Reply:
Calling the `RegisterParametersFor` should be enough, no? Parameterized test discovery was implemented prior to this PR without the need for a marker interface.

Reply:
if a test class implements that interface then for each test target the class is invoked with the target to exec the default value extraction